lablib/renderers/basic.py

```diff
@@ -1,149 +1,327 @@
 from __future__ import annotations
-from dataclasses import dataclass
+from dataclasses import dataclass, field
 
-import subprocess
+import logging
 import shutil
-from typing import List
+import tempfile
+from typing import Any, Dict, List, Optional, Set, Union
 
 from pathlib import Path
 
-from ..generators import OCIOConfigFileGenerator
-from ..processors import OIIORepositionProcessor
-from ..lib import SequenceInfo
+from ..lib import SequenceInfo, utils
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+SUPPORTED_CODECS = ["ProRes422-HQ", "ProRes4444-XQ", "DNxHR-SQ"]
 
 
 @dataclass
-class BasicRenderer:
-    color_proc: OCIOConfigFileGenerator = None
-    repo_proc: OIIORepositionProcessor = None
-    source_sequence: SequenceInfo = None
-    staging_dir: str = None
-    name: str = None
-    format: str = None
+class Codec:
+    name: str = field(default_factory=str, init=True, repr=True)
+
+    def __post_init__(self) -> None:
+        if self.name not in SUPPORTED_CODECS:
+            raise ValueError(
+                f"{self.name} is not found in supported codecs.\n{SUPPORTED_CODECS = }"
+            )
+
+    def get_ffmpeg_args(self) -> List[str]:
+        args = []
+        # fmt: off
+        # TODO: i should probably abstract the cmdargs
+        if self.name == "ProRes422-HQ":
+            args = [
+                "-vcodec", "prores_ks",
+                "-profile:v", "3",
+                "-vendor", "apl0",
+                "-pix_fmt", "yuv422p10le",
+                "-vtag", "apch",
+            ]
+        if self.name == "ProRes4444-XQ":
+            args = [
+                "-vcodec", "prores_ks",
+                "-profile:v", "4",
+                "-vendor", "apl0",
+                "-pix_fmt", "yuva444p10le",
+                "-vtag", "ap4h",
+            ]
+        if self.name == "DNxHR-SQ":
+            args = [
+                "-vcodec", "dnxhd",
+                "-profile:v", "2",
+                "-pix_fmt", "yuv422p",
+            ]
+        # fmt: on
+
+        return args
+
+
+@dataclass
+class Burnin:
+    data: Dict[str, str] = field(default_factory=dict)
+
+    size: int = field(default=64)
+    padding: int = field(default=30)
```

Comment:
It doesn't look this is used anywhere, right?

Reply:
true the `padding` is not yet used anywhere. i forgot to add it back in when trying out the font paths.
will add the padding back to the burnins